src/data/rpc.js

```diff
@@ -260,6 +290,11 @@ function sendObjRequest(objOrTsid, fname, args, callback) {
  */
 function sendRequest(gsid, rpcFunc, args, callback) {
 	assert(gsid !== config.getGsid(), 'RPC to self');
+	if (shuttingDown) {
+		// hack - see above (preShutdown)
+		log.info('shutdown in progress, dropping %s RPC request', rpcFunc);
+		return;
+	}
```

Comment:
What if some of the pending requests we're processing should make outgoing rpc requests?